### ISSUE TITLE
Revert workaround for https://github.com/dotnet/roslyn/issues/3898

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#1520]
https://github.com/dotnet/roslyn-analyzers/pull/1519 introduced a temporary reflection based workaround for https://github.com/dotnet/roslyn/issues/3898. This reflection based code must be removed once the underlying Roslyn bug is fixed.

**Comment by mavasani** [comment]
@drewnoakes The fix was in 2.7.X MS.CA package, and the master branch is still on 2.6. I have [already fixed](https://github.com/dotnet/roslyn-analyzers/blob/features/dataflow/src/Microsoft.NetCore.Analyzers/Core/Runtime/SerializationRulesDiagnosticAnalyzer.cs#L265) this issue in features/dataflow branch, which is on 2.9 package. I think the resolution here depends on whether we want to move master forward to 2.7.X or directly to 2.9.X/3.0.X by merging back features/dataflow, once the dataflow branch is ready to ship analyzers (it is based primarily on CFG APIs that went public without a feature flag recently).

**Comment by mavasani** [comment]
No longer relevant, closing.
